### PR TITLE
added api blueprint

### PR DESCRIPTION
**Pull request by johncookds**
This is the api blueprint. I tried to make it match already completed work in the controllers while matching as much as I could the issue request. The get_relations and get_events routes may need editing once the appropriate controllers are written.

To solve issue: https://github.com/arXiv/arxiv-external-links/issues/12

### CODE REVIEW

**Comment by bonotake** [comment]
Hi @johncookds,

I think you should add tests. You can see:
https://github.com/arXiv/arxiv-zero/blob/master/zero/routes/tests/test_external_api.py is a sample test code of `routes` in a sample project arxiv-zero.

Moreover, there are several points that are not compliant with PEP008. Try `pipenv run pylint relations/routes/api.py`.

**Comment by johncookds** [comment]
Ah, great thanks for the link and pylint info. I will close this pull request and re-open once I have tests and its passing the pylint.

File: relations/routes/api.py [hunk 1]
```diff
@@ -21,3 +21,43 @@ def service_status() -> Response:
     response.status_code = status_code
     response.headers.extend(headers)
     return response
+
+@blueprint.route('/{str:arxiv_id_str}v{int:arxiv_ver}/relations',methods=['POST'])
+def create_new(arxiv_id_str: str,arxiv_ver: int) -> Response:
+    response_data,status_code,headers = controllers.create_new(arxiv_id_str,arxiv_ver,request.params)
```
Comment (bonotake) [inline relations/routes/api.py:27]: It doesn't match the return values of `controllers.create_new`. Please check below. Plus, the function may raise an `InternalServerError` exception.
https://github.com/arXiv/arxiv-external-links/blob/e29c191b0c68be9ac72cfbe570936dcfbe57e1bd/relations/controllers.py#L49-L58

Moreover, I am not sure whether you can return all of the `Relation` values to users.

File: relations/routes/api.py [hunk 2]
```diff
@@ -21,3 +21,43 @@ def service_status() -> Response:
...
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/{str:arxiv_id_str}v{int:arxiv_ver}/relations/{str:relation_id_str}',methods=['PUT'])
+def supercede(arxiv_id_str: str,arxiv_ver: int,relation_id_str: str) -> Response:
+    response_data,code,headers = controllers.supercede(arxiv_id_str,arxiv_ver,relation_id_str,request.params)
```
Comment (bonotake) [inline relations/routes/api.py:35]: ditto
https://github.com/arXiv/arxiv-external-links/blob/e29c191b0c68be9ac72cfbe570936dcfbe57e1bd/relations/controllers.py#L107-L114

File: relations/routes/api.py [hunk 3]
```diff
@@ -21,3 +21,43 @@ def service_status() -> Response:
...
+    response: Response = jsonify(response_data)
+    response.status_code = status_code
+    response.headers.extend(headers)
+    return response
+
+@blueprint.route('/{str:arxiv_id_str}v{int:arxiv_ver}/relations/{str:relation_id_str}',methods=['DELETE'])
+def suppress(arxiv_id_str: str,arxiv_ver: int,relation_id_str: str) -> Response:
```
Comment (bonotake) [inline relations/routes/api.py:42]: ditto
https://github.com/arXiv/arxiv-external-links/blob/e29c191b0c68be9ac72cfbe570936dcfbe57e1bd/relations/controllers.py#L172-L179

**Review by bonotake — COMMENTED** [review]
Though this PR is now closed, I did a brief review.

**Pull request closed without merging**